src/test/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateStepTest.java

```diff
@@ -0,0 +1,205 @@
+import hudson.model.queue.QueueTaskFuture;
+import hudson.util.ReflectionUtils;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep.DescriptorImpl;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static hudson.model.Result.FAILURE;
+import static java.lang.String.format;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit tests for {@link ExwsAllocateStep}.
+ *
+ * @author Alexandru Somai
+ */
+public class ExwsAllocateStepTest {
+
+    private static final String DISK_POOL_ID = "disk-pool-1";
+    private static final String DISK_ID = "disk1";
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private WorkflowRun upstreamRun;
+    private WorkflowRun downstreamRun;
+
+    /* ##### Tests for the upstream Job ###### */
+
+    @Test
+    public void missingDiskPoolId() throws Exception {
+        createUpstreamJobAndRun(" ");
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains("Disk Pool ID was not provided as step parameter", upstreamRun);
+    }
+
+    @Test
+    public void wrongDiskPoolId() throws Exception {
+        setupDiskPool();
+        createUpstreamJobAndRun(DISK_POOL_ID + "random");
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disk Pool ID matching '%s' was found in the global config", DISK_POOL_ID + "random"), upstreamRun);
+    }
+
+    @Test
+    public void emptyDisks() throws Exception {
+        setupDiskPool();
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disks were defined in the global config for Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingDiskId() throws Exception {
+        setupDiskPool(new Disk("", "name", "mount", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Disk ID was not provided in the Jenkins global config for the Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingMasterMountPoint() throws Exception {
+        setupDiskPool(new Disk(DISK_ID, "name", "", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Mounting point from Master to the disk is not defined for Disk ID '%s', from Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
```

Comment:
I think it is okay to have the `missingMasterMountPoint` test here for now, but when there is more than one allocation strategy, it may be better to place strategy testing in a test file of its own. At this point, no change is needed IMO.